### PR TITLE
feat: add init-reserve-management smart contract execution

### PR DESCRIPTION
**Pull request by LGLO**
# Description

Requires follow ups:
* unit tests for transaction properties, ones written in the header of init.rs file
* implementation of idempotency
* implement command

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages.
- [X] New tests are added if needed and existing tests are updated. <- partly
- [ ] Relevant logging and metrics added
- [x] CI passes. See note on CI.
- [ ] Any changes are noted in the `changelog.md` for affected crate <- this is not exposed to users yet
- [x] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG Partner Chains developers to do this
for you.

### CODE REVIEW

File: toolkit/offchain/src/csl.rs [hunk 1]
```diff
@@ -184,15 +184,15 @@ impl ScriptExUnits {
 
 pub(crate) fn get_validator_budgets(
 	mut responses: Vec<OgmiosEvaluateTransactionResponse>,
-) -> Result<ScriptExUnits, JsError> {
```
Comment (LGLO) [inline toolkit/offchain/src/csl.rs:187]: There is nothing to fail in this method

File: toolkit/offchain/src/reserve/mod.rs [hunk 2]
```diff
@@ -0,0 +1,3 @@
+//! All smart-contracts related to rewads token reserve management
```
Comment (LGLO) [inline toolkit/offchain/src/reserve/mod.rs:1]: I've locally fixed this typo already. Sparing CI run for it.

File: toolkit/offchain/src/reserve/init.rs [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+) -> anyhow::Result<Vec<McTxHash>> {
+	let reserve_validator =
+		ScriptData::new("Reserve Management Validator", RESERVE_VALIDATOR.to_vec(), 28);
+	let reserve_policy =
+		ScriptData::new("Reserve Management Policy", RESERVE_AUTH_POLICY.to_vec(), 29);
+	let ics_validator = ScriptData::new(
+		"Illiquid Circulation Validator",
+		ILLIQUID_CIRCULATION_SUPPLY_VALIDATOR.to_vec(),
+		30,
+	);
```
Comment (kpinter-iohk) [inline toolkit/offchain/src/reserve/init.rs:60]:
```suggestion
	let reserve_validator =
		ScriptData::new("Reserve Management Validator", RESERVE_VALIDATOR.to_vec(), raw_scripts::ScriptId::ReserveValidator as u16);
	let reserve_policy =
		ScriptData::new("Reserve Management Policy", RESERVE_AUTH_POLICY.to_vec(), raw_scripts::ScriptId::ReserveAuthPolicy as u16);
	let ics_validator = ScriptData::new(
		"Illiquid Circulation Validator",
		ILLIQUID_CIRCULATION_SUPPLY_VALIDATOR.to_vec(),
		raw_scripts::ScriptId::IlliquidCirculationSupplyValidator as u16,
	);
```

Comment (LGLO) [reply toolkit/offchain/src/reserve/init.rs:60]: Good one, I'll change the type in ScriptData, to reduce casts

**Review by kpinter-iohk — APPROVED** [review]

**Pull request merged**